kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -395,8 +395,13 @@ object Channel:
         final class ZeroCapacityUnsafe[A](using initFrame: Frame, allow: AllowUnsafe) extends BaseUnsafe[A]:
             val isClosed = AtomicBoolean.Unsafe.init(false)
 
-            private val closedResult                     = Result.fail(Closed("Channel", initFrame, "zero-capacity"))
-            def asResult[A](value: A): Result[Closed, A] = if isClosed.get() then closedResult else Result.succeed(value)
+            private val closedResult = Result.fail(Closed("Channel", initFrame, "zero-capacity"))
+
+            private def asResult[A](value: A): Result[Closed, A] =
+                if isClosed.get() then closedResult else Result.succeed(value)
+
+            private def asOptimisticResult[A](value: Chunk[A]): Result[Closed, Chunk[A]] =
```

Comment:
The name doesn't seem very informative, how about `asDrainResult`? Could you also add a comment explaining the edge case the method handles?